### PR TITLE
fix: handle results of polling updates in allocation actor

### PR DESCRIPTION
**Pull request by stoksc**
## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/internal/task/allocation.go [hunk 1]
```diff
@@ -553,6 +553,10 @@ func (a *Allocation) ResourcesStateChanged(
 	ctx.Log().Debugf("resources state changed: %+v", msg)
 	switch msg.ResourcesState {
 	case sproto.Pulling:
+		if !model.LessAllocationState(a.getModelState(), model.AllocationStatePulling) {
```
Comment (stoksc) [inline master/internal/task/allocation.go:556]: this is basically identical to the change on line 575

Comment (jerryharrow) [reply master/internal/task/allocation.go:556]: Interesting.   This stuff only gets set on Pulling?   Before Rig added that state relatively recently, we went directly to Running.    There is actually no guarantee that we will see a Pulling state -- it depends upon timing.
Is that OK?

**Review by jerryharrow — APPROVED** [review]

File: master/internal/task/allocation.go (hunk 1 above)
Comment (stoksc) [reply master/internal/task/allocation.go:556]: oh, goodness. ok, we should chat.

**Review by jerryharrow — APPROVED** [review]

File: master/internal/task/allocation.go (hunk 1 above)
Comment (stoksc) [reply master/internal/task/allocation.go]: ok, fixed it to handle this

**Pull request merged**